bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/DecimalValue.java

```diff
@@ -72,7 +72,8 @@ public DecimalValue(String value) {
                 this.value = new BigDecimal(value, MathContext.DECIMAL128);
             } catch (NumberFormatException exception) {
                 String message = exception.getMessage();
-                if (message.equals("Too many nonzero exponent digits.") || message.equals("Exponent overflow.")) {
+                if ((message != null) && (message.equals("Too many nonzero exponent digits.") ||
```

Comment:
Should we handle empty string separately instead of replying on message != null

Reply:
No, If the received exception message is not equal to those two strings it will anyway `throw exception;`. The problem was `exception.getMessage();` sometimes may be null, and then couldn't use `equals` method on `null`.